Goroutine leak: EXT-X-PRELOAD-HINT handler returns while holding the muxer mutex when the stream is closed (LL-HLS)

### Version

`github.com/bluenviron/gohlslib/v2 v2.4.0`
Variant: `MuxerVariantLowLatency`

### Summary

The EXT-X-PRELOAD-HINT part handler in `muxer_stream.go` acquires `s.mutex` and waits on
`s.cond`. When the muxer is closed while a preload-hint request is pending, the handler wakes
up, sees `s.closed`, and returns **without unlocking the mutex**.

Since `muxerStream.mutex` is a pointer to the muxer-wide `Muxer.mutex`
(`muxer.go`: `mutex sync.Mutex`, `m.cond = sync.NewCond(&m.mutex)`), every subsequent goroutine
that needs that mutex — in-flight playlist/part handlers of the same muxer — blocks forever.
It is a silent goroutine leak rather than a crash, so it accumulates in deployments where
muxers are torn down and recreated (source restarts, config reloads, idle reclaim).

Note that the other two handlers in the same file use `defer s.mutex.Unlock()`; only this one
manages the lock manually (it has to, because of `cond.Wait`), and the `closed` early-return
path misses the unlock.

### Relevant code

`muxer_stream.go` (v2.4.0, unchanged on `main`):

```go
// EXT-X-PRELOAD-HINT
capturePartID := s.nextPartID
partPath := partPath(s.prefix, s.id, capturePartID)
s.server.registerPath(
    partPath,
    func(w http.ResponseWriter, r *http.Request) {
        s.mutex.Lock()

        for {
            if s.closed {
                w.WriteHeader(http.StatusInternalServerError)
                return                     // <-- returns while still holding s.mutex
            }

            if s.nextPartID > capturePartID {
                break
            }

            s.cond.Wait()
        }

        h := s.server.getPathHandler(partPath)

        s.mutex.Unlock()

        if h != nil {
            h(w, r)
        }
    })
```

`Muxer.Close()` broadcasts on the condition variable to wake pending handlers, which is exactly
what triggers the faulty return path — so closing the muxer is what leaks the lock.

### Reproduction

Self-contained test (reuses the existing helpers `doRequest`, `testVideoTrack`, `testH264SPS`, `testTime`):

```go
package gohlslib

import (
	"testing"
	"time"

	"github.com/stretchr/testify/require"
)

// 1. start an LL-HLS muxer and produce a few parts
// 2. request the preload-hint part (handler blocks in cond.Wait while holding the mutex)
// 3. Close() the muxer  -> handler wakes, sees s.closed, returns WITHOUT unlocking
// 4. any further request on the same muxer blocks forever on that mutex
func TestPreloadHintLockLeak(t *testing.T) {
	m := &Muxer{
		Variant:            MuxerVariantLowLatency,
		SegmentCount:       7,
		SegmentMinDuration: 1 * time.Second,
		Tracks:             []*Track{testVideoTrack},
	}
	require.NoError(t, m.Start())

	for i := 1; i <= 3; i++ {
		d := time.Duration(i) * time.Second
		err := m.WriteH264(testVideoTrack, testTime.Add(d-1*time.Second),
			int64(d)*int64(testVideoTrack.ClockRate)/int64(time.Second),
			[][]byte{testH264SPS, {8}, {5}})
		require.NoError(t, err)
	}

	mv, _, err := doRequest(m, "index.m3u8")
	require.NoError(t, err)

	mediaPath := ""
	for _, line := range splitLines(string(mv)) {
		if indexOf(line, "_stream.m3u8") >= 0 && indexOf(line, "URI=") < 0 {
			mediaPath = line
		}
	}
	require.NotEmpty(t, mediaPath)

	mb, _, err := doRequest(m, mediaPath)
	require.NoError(t, err)

	hint := ""
	for _, line := range splitLines(string(mb)) {
		if indexOf(line, "EXT-X-PRELOAD-HINT") >= 0 {
			if u := extractURI(line); u != "" {
				hint = u
			}
		}
	}
	require.NotEmpty(t, hint)

	// pending preload-hint request
	hintDone := make(chan struct{})
	go func() {
		defer close(hintDone)
		_, _, _ = doRequest(m, hint)
	}()
	time.Sleep(300 * time.Millisecond)

	m.Close()

	select {
	case <-hintDone:
	case <-time.After(2 * time.Second):
		t.Fatal("preload hint handler did not return")
	}

	// any request needing the muxer mutex now hangs
	after := make(chan struct{})
	go func() {
		defer close(after)
		_, _, _ = doRequest(m, mediaPath)
	}()

	select {
	case <-after:
		// fixed
	case <-time.After(3 * time.Second):
		t.Fatal("request blocked forever: mutex was never released by the preload-hint handler")
	}
}

// --- helpers used above ---

func splitLines(s string) []string {
	var out []string
	cur := ""
	for _, r := range s {
		if r == '\n' {
			out = append(out, cur)
			cur = ""
			continue
		}
		if r != '\r' {
			cur += string(r)
		}
	}
	if cur != "" {
		out = append(out, cur)
	}
	return out
}

func indexOf(s, sub string) int {
	for i := 0; i+len(sub) <= len(s); i++ {
		if s[i:i+len(sub)] == sub {
			return i
		}
	}
	return -1
}

func extractURI(line string) string {
	i := indexOf(line, `URI="`)
	if i < 0 {
		return ""
	}
	rest := line[i+5:]
	for j := 0; j < len(rest); j++ {
		if rest[j] == '"' {
			return rest[:j]
		}
	}
	return ""
}
```

Result on v2.4.0:

```
--- FAIL: TestPreloadHintLockLeak (3.30s)
    request blocked forever: mutex was never released by the preload-hint handler
```

### Suggested fix

Release the mutex on the `closed` path:

```diff
 for {
     if s.closed {
+        s.mutex.Unlock()
         w.WriteHeader(http.StatusInternalServerError)
         return
     }
```

With this one-line change the test above passes and the full existing test suite still passes
(`go test ./...` → ok).

--- a/muxer_stream.go
+++ b/muxer_stream.go
@@ -730,6 +730,7 @@ func (s *muxerStream) rotateParts(
 
 				for {
 					if s.closed {
+						s.mutex.Unlock()
 						w.WriteHeader(http.StatusInternalServerError)
 						return
 					}

